[Bug 21086] Keep correct selection when formatting script

BerndN opened the pull request:
http://quality.livecode.com/show_bug.cgi?id=21086

BerndN committed:
[Bug 21086] Keep correct selection when formatting script

http://quality.livecode.com/show_bug.cgi?id=21086

BerndN committed:
Create bugfix-21086.md

montegoulding commented on Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript:
```diff
@@ -2992,11 +2992,33 @@ command scriptFormat pScope, pDontGroup
+      if word 4 of it is a number then put word 4 of it into tLastLine else put tFirstLine into tLastLine
+      
+      put the number of chars of line 1 to tFirstLine of tOldText into tCharsAboveFirstLineOld
+      put the number of chars of line 1 to tLastLine of tOldText into tCharsAboveLastLineOld
```
@BerndN this is likely to be slow for long scripts. Could you add something like:
```
if tFirstLine is tLastLine then
    put tCharsAboveFirstLineOld into tCharsAboveLastLineOld
else
    put the number of chars of line 1 to tLastLine of tOldText into tCharsAboveLastLineOld
end if
```
Same below

Also it might be worth experimenting with `charIndex` for speed:

`put the charIndex of line tFirstLine of tOldText`

BerndN committed:
Update revsecommoneditorbehavior.livecodescript

added testing for if tFirstLine is tLastLine, though the speed difference is not measurable, both version take 12 milliseconds on a 3600 line script. But I agree it is a cleaner way to do it.

Additionally I changed 
put (the number of chars of line 1 to tFirstLine - 1 of tNewText) + 1 into tFirstLineChar
from + 2 to +1. This also keeps the selection if the cursor is positioned before the first char.

As far as I understand charIndex works only on fields not variables. I did test and there is not speed improvement using charIndex on the field.

montegoulding commented:
@livecode-vulcan review ok afd8db2

livecode-vulcan commented:
💙 review by @montegoulding ok afd8db2db4a1e871050de3d2774403e0d47ce02b
